test/test_sparse_csr.py

```diff
@@ -0,0 +1,202 @@
+import torch
+
+torch.set_default_dtype(torch.double)
+
+import functools
+import random
+import operator
+import numpy as np
+import warnings
+from torch.testing._internal.common_utils import TestCase, run_tests, load_tests
+
+# load_tests from torch.testing._internal.common_utils is used to automatically filter tests for
+# sharding on sandcastle. This line silences flake warnings
+load_tests = load_tests
+
+class TestSparseCSR(TestCase):
+    def gen_sparse_csr(self, shape, nnz):
+        total_values = functools.reduce(operator.mul, shape, 1)
+        dense = np.random.randn(total_values)
+        fills = random.sample(list(range(total_values)), total_values - nnz)
+
+        for f in fills:
+            dense[f] = 0
+        dense = torch.from_numpy(dense.reshape(shape))
+
+        return dense.to_sparse_csr()
```

Comment:
This seems like a horribly inefficient way of sampling random CSR tensors, since you have to materialize an entire dense tensor first before converting it to CSR...